code/modules/virus2/effect/effect.dm

```diff
@@ -559,6 +559,44 @@
 	mob.dna.SetSEState(VEGANBLOCK,1)
 	domutcheck(mob, null)
 
+/datum/disease2/effect/famine
+	name = "Faminous Potation"
+	stage = 2
+	max_multiplier = 3
+
+/datum/disease2/effect/famine/activate(var/mob/living/carbon/mob)
+	if(ishuman(mob))
+		var/mob/living/carbon/human/H = mob
+		if(H.dna)
+			if(H.species.flags & IS_PLANT) //Plantmen take a LOT of damage
+				H.adjustCloneLoss(5 * multiplier)
+
+	for(var/obj/machinery/portable_atmospherics/hydroponics/H in range(3*multiplier,mob))
+		if(H.seed && !H.dead) // Get your xenobotanist/vox trader/hydroponist mad with you in less than 1 minute with this simple trick.
+			switch(rand(1,3))
```

Comment:
```dm
var/effect = rand(1, 3)
switch(effect)
     if (DECREASE_NUTRIMENT)
         ....
```
Might be me being a brainlet but I find it a bit more readable this way. Your call.